tiledb/api/src/array/mod.rs

```diff
@@ -121,39 +121,45 @@ impl TryFrom<ffi::tiledb_layout_t> for CellOrder {
     }
 }
 
-pub(crate) struct RawArray {
-    ffi: *mut ffi::tiledb_array_t,
+pub enum RawArray {
+    Owned(*mut ffi::tiledb_array_t),
+    Borrowed(*mut ffi::tiledb_array_t),
```

Comment:
After finishing my review, I don't remember these being used? Grabbing a copy of a pointer that's not freed really makes me worry that we'll have things like the original pointer being dropped out from under us. I'm not (yet) good enough with Rust's borrow rules to know if the compiler would hell there, but I'm guessing not since its a pointer copy?

Reply:
I think these existed as an intermediate state before I spent a bit more time with `#[derive(QueryCAPIInterface)]`.  Will remove.  Fully agree that these would need a pretty hefty comment about safety, the problem you describe is absolutely a possibility.